temp_streaming_function.py

```
@@ -0,0 +1,394 @@
+                        for persona_id, agent in orchestrator.persona_agents.items():
+                            print(f"[DEBUG] Clearing conversation history for existing agent: {persona_id}")
+                            agent.clear_conversation_history(user_progress.id)
+                            print(f"Cleared conversation history for existing persona agent: {persona_id}")
+                elif orchestrator._last_scene_id is None:
+                    print(f"[DEBUG] First time setting _last_scene_id to: {current_scene_id}")
+                
+                # Store current scene ID for next comparison
+                orchestrator._last_scene_id = current_scene_id
+            
+            current_scene = orchestrator.scenario.get('scenes', [{}])[orchestrator.state.current_scene_index]
+            timeout_turns = current_scene.get('timeout_turns') or current_scene.get('max_turns', 15)
```

Ruff (0.14.1)

Local variable `timeout_turns` is assigned to but never used

Remove assignment to unused variable `timeout_turns`

(F841)